internal/notification/pubsub/pubsub.go

```diff
@@ -0,0 +1,136 @@
+package pubsub
+
+import (
+	"context"
+	"encoding/json"
+	"net/url"
+	"regexp"
+
+	cloudpubsub "cloud.google.com/go/pubsub"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/internal/health"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/internal/util"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/run/v1"
+)
+
+// Client represents a client to Google Cloud Pub/Sub.
+type Client interface {
+	Publish(ctx context.Context, event RolloutEvent) error
+}
+
+// PubSub is a Google Cloud Pub/Sub client to publish messages.
+type PubSub struct {
+	topic *cloudpubsub.Topic
+}
+
+// RolloutEvent is the format of an event published to Pub/Sub.
+type RolloutEvent struct {
+	Event                        string       `json:"event"`
+	CandidateRevisionName        string       `json:"candidateRevisionName"`
+	CandidateRevisionPercent     int          `json:"candidateRevisionPercent"`
+	CandidateRevisionURL         string       `json:"candidateRevisionURL"`
+	CandidateWasPromotedToStable bool         `json:"candidateWasPromotedToStable"`
+	Service                      *run.Service `json:"service"`
+}
+
+// New initializes a PubSub client to a topic in a project.
+func New(ctx context.Context, projectID string, topicName string) (ps PubSub, err error) {
```

Comment:
this should return a `func()` to "stop or wait flushing"

Reply:
Should we return a third value or can we just add a wrapper for `Stop` in our package?